File: 2022/02/24/addDigits.ts
/**
 * https://leetcode.com/problems/add-digits/
 * @author Andrew Jarombek
 * @since 2/24/2022
 */

function addDigits(num: number): number {
    while (num > 9) {
        let n = 0;
        let m = num;

        while (m > 0) {
            n += m % 10;
            m = Math.floor(m / 10);
        }

        num = n;
    }

    return num;
}

File: 2022/07/11/java/AddDigits.java
/**
 * https://leetcode.com/problems/add-digits
 * @author Andrew Jarombek
 * @since 7/11/2022
 */

public class AddDigits {
    public int addDigits(int num) {
        while (num > 9) {
            int val = num;
            int next = 0;

            while (val > 9) {
                next += val % 10;
                val /= 10;
            }

            next += val;
            num = next;
        }

        return num;
    }

    public static void main(String[] args) {
        var program = new AddDigits();
        assert program.addDigits(12) == 3;
        assert program.addDigits(1) == 1;
    }
}
